src/main/java/org/jenkinsci/plugins/workflow/flow/StashManager.java

```diff
@@ -83,20 +89,35 @@ public static void stash(@Nonnull Run<?,?> build, @Nonnull String name, @Nonnull
         stash(build, name, workspace, listener, includes, excludes, useDefaultExcludes, false);
     }
 
+    @Deprecated
+    public static void stash(@Nonnull Run<?,?> build, @Nonnull String name, @Nonnull FilePath workspace, @Nonnull TaskListener listener,
+                             @CheckForNull String includes, @CheckForNull String excludes, boolean useDefaultExcludes, boolean allowEmpty) throws IOException, InterruptedException {
+        stash(build, name, workspace, launcherFor(workspace, listener), envFor(build, workspace, listener), listener, includes, excludes, useDefaultExcludes, allowEmpty);
+    }
+
     /**
      * Saves a stash of some files from a build.
      * @param build a build to use as storage
      * @param name a simple name to assign to the stash (must follow {@link Jenkins#checkGoodName} constraints)
      * @param workspace a directory to use as a base
+     * @param launcher a way to launch processes, if required
+     * @param env environment to use when launching processes, if required
+     * @param listener a way to report progress or problems
      * @param includes a set of Ant-style file includes, separated by commas; null/blank is allowed as a synonym for {@code **} (i.e., everything)
      * @param excludes an optional set of Ant-style file excludes
      * @param useDefaultExcludes whether to use Ant default excludes
      * @param allowEmpty whether to allow an empty stash
+     * @see StashAwareArtifactManager#stash
      */
     @SuppressFBWarnings(value="RV_RETURN_VALUE_IGNORED_BAD_PRACTICE", justification="fine if mkdirs returns false")
-    public static void stash(@Nonnull Run<?,?> build, @Nonnull String name, @Nonnull FilePath workspace, @Nonnull TaskListener listener,
+    public static void stash(@Nonnull Run<?,?> build, @Nonnull String name, @Nonnull FilePath workspace, @Nonnull Launcher launcher, @Nonnull EnvVars env, @Nonnull TaskListener listener,
```

Comment:
@abayer FYI, this is the new overload which could plausibly be combined with a `boolean verbose` option to address [JENKINS-40912](https://issues.jenkins-ci.org/browse/JENKINS-40912) in lieu of #62. A minor API change like that could be done in trunk at low risk, leaving the introduction of `StashAwareArtifactManager` and core baseline changes to this PR for later.

Reply:
Not entirely clear on what you mean? I made a comment over at https://issues.jenkins-ci.org/browse/JENKINS-40912 wondering whether building support for listing stash files into `StashAwareArtifactManager` makes sense, since anything we do here like #62 would break horribly the moment the `StashAwareArtifactManager` change came in... I dunno. Everything about JENKINS-40912 makes me want to light myself on fire at this point. =)

Reply:
Responded in JIRA.

Reply:
Possible, though six are mandatory and only five optional, and there is only one caller so we are not really simplifying anything that way. Option builders are useful when you have an open-ended list of callers and most of them do not need to specify most of the options.

Reply:
I'll let it go for now -- you're right that the set of callers is currently pretty limited.